lib/matplotlib/patches.py

```diff
@@ -3308,7 +3308,9 @@ def transmute(self, path, mutation_size, linewidth):
             x0, y0 = path.vertices[0]
             x1, y1 = path.vertices[1]
 
-            if self.beginarrow:
+            # If there is no room for an arrow and a line, then skip the arrow
+            hasBeginArrow = self.beginArrow and not ((x0==x1) and (y0==y1))
```

Comment:
spelling error, should be lower case `beginarrow`

Can you use `has_begin_arrow` for the local variable instead of camel case please?

Probably the same thing with `endarrow`